kyo-core/shared/src/main/scala/kyo/ios.scala

```diff
@@ -139,9 +139,9 @@ sealed trait IOs extends Effect[IOs]:
                             ensureLoop(v2, np)
                         end apply
                 case _ =>
-                    p.remove(ensure)
                     IOs {
                         ensure()
+                        p.remove(ensure)
```

Comment:
It seems the ordering here could make the `ensure` get lost. The ensure would be removed but then the `IOs` instance might never get executed if the execution is interrupted.